### PR TITLE
subunit: 1.1.0 -> 1.3.0

### PR DESCRIPTION
**Pull request by drewrisinger**
###### Motivation for this change

Bump subunit as requirement for tester (``stestr``) for upcoming qiskit package #78772

launchpad fetch doesn't work for >1.1.0, so switched to equivalent git repo.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drewrisinger** [comment]
@domenkozar seems to be the owner. No maintainer listed.

**Review by jonringer — CHANGES_REQUESTED** [review]
fails:
```
builder for '/nix/store/iz7xra43k6630qd4xwrj21xc6j42lfnj-subunit-1.3.0.drv' failed with exit code 2; last 10 log lines:
  unpacking source archive /nix/store/abmxrfp9rvprjgi8j7f5jjpbc3df2fgz-source
  source root is source
  patching sources
  configuring
  no configure script, doing nothing
  building
  no Makefile, doing nothing
  installing
  install flags: SHELL=/nix/store/715fljdrspphbdk255r729zxp4w7lcgk-bash-4.4-p23/bin/bash install
  make: *** No rule to make target 'install'.  Stop.
```

File: pkgs/development/libraries/subunit/default.nix [hunk 1]
```diff
@@ -1,17 +1,19 @@
-{ stdenv, fetchurl, pkgconfig, check, cppunit, perl, pythonPackages }:
+{ stdenv, fetchFromGitHub, pkg-config, check, cppunit, perl, pythonPackages }:
```
Comment (jonringer) [inline pkgs/development/libraries/subunit/default.nix]: needs autotools
```suggestion
{ stdenv, fetchFromGitHub, pkg-config, autoreconfHook, check, cppunit, perl, pythonPackages }:
```

File: pkgs/development/libraries/subunit/default.nix [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
-    sha256 = "0lcah7p66c05p7xpw6ns1is0i02lh0nq8gq51mv4wyvbr6zaasa8";
+  src = fetchFromGitHub {
+    owner = "testing-cabal";
+    repo = pname;
+    rev = version;
+    sha256 = "1hk5v8gg5azwrc079xqigj0sp6qm47vzg1qwjbvb8dbjwlg75rvz";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ pkg-config ];
```
Comment (jonringer) [inline pkgs/development/libraries/subunit/default.nix]: enable creation of Makefile
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config ];
```

**Review by jonringer — CHANGES_REQUESTED** [review]
the python package needs some love:

```
builder for '/nix/store/gzn71xb2k3iqs8mv6b4i4a0i3a7iy22l-python3.7-subunit-1.3.0.drv' failed with exit code 1; last 10 log lines:
  WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
  WARNING: The directory '/homeless-shelter/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
  WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff72c6490>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/hypothesis/
  WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5e9bad0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/hypothesis/
  WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5e9bcd0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/hypothesis/
  WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5e9be90>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/hypothesis/
  WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ffff5eab090>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/hypothesis/
  ERROR: Could not find a version that satisfies the requirement hypothesis (from versions: none)
  ERROR: No matching distribution found for hypothesis
```

**Comment by drewrisinger** [comment]
Sorry for the delay in this, looked like tests weren't actually being run. So trying to fix those, and running into a lot of issues. I wanted to bring them up to https://github.com/testing-cabal/subunit, but that page doesn't have an issues section, just PRs, and I don't have the time to debug those deep errors (looks like tests haven't been updated for years).

**Comment by jonringer** [comment]
you don't have to run all the tests. I just want some code coverage to ensure the package is actually installed correctly.

Otherwise installing a wheel is essentially: "I dumped a bumped of text into a directory, with no guarantee it works".

**Comment by drewrisinger** [comment]
I'm going to let this lapse. I don't have the time/interest to devote to getting this & its tests running properly. Thought I needed this for a ``pythonPackages.stestr`` package, but don't need that one anymore.

**Pull request closed without merging**